projects/packages/sync/src/class-sender.php

```diff
@@ -269,6 +270,12 @@ public function do_full_sync() {
 			return new \WP_Error( 'jetpack_sync_read_only' );
 		}
 
+		// Sync not started or Sync finished.
+		$status = $sync_module->get_status();
+		if ( false === $status['started'] || ( ! empty( $status['started'] ) && ! empty( $status['finished'] ) ) ) {
```

Comment:
Thanks I appreciate the feedback. However I am going to ignore for the moment on the fact that I don't want to introduce further changes to this PR. There are several re-uses of functions and options that need clean up without the Full Sync module and I think it would be worthwhile to do a full overhaul. 

The main context I'm using in this decision is that we are trying to move away from using Full Sync so less touches on it the better :)

You are correct this check happens later on the goal is to move it above the locking and option updates to remove unnecessary rights to the database when no Full Sync is in progress.

But I'm guessing if continue_enqueuing() did nothing then the queue will be empty? 
Yep :)